### ISSUE TITLE
This query of the stats db took down our master

### ISSUE THREAD

**Issue by karlgrz** [rethinkdb/rethinkdb#5327]
```
r.db("rethinkdb").table("stats").get(["database", "events"])
```

We're running a 2 node cluster right now, and that query crashed our master. What kind of logs can I attach to help out? That db has a one table with about 500k records in it.

RethinkDB 2.2.3-1

**Comment by karlgrz** [comment]
I -think- this is the trace for the query:

```
2016-01-28T15:18:26.564162880 29.624563s error: Thu Jan 28 15:18:25 2016\n\n1 [0xb03bf3]: format_backtrace(bool) at 0xb03bf3 (rethinkdb)\n2 [0xda5944]: report_fatal_error(char const*, int, char const*, ...) at 0xda5944 (rethi
nkdb)\n3 [0xa10343]: linux_thread_pool_t::fatal_signal_handler(int, siginfo*, void*) at 0xa10343 (rethinkdb)\n4 [0x7fd3288c3cb0]: /lib/x86_64-linux-gnu/libpthread.so.0(+0xfcb0) [0x7fd3288c3cb0] at 0x7fd3288c3cb0 (/lib/x86_64-
linux-gnu/libpthread.so.0)\n5 [0xc5a068]: stats_artificial_table_backend_t::read_row(ql::datum_t, signal_t*, ql::datum_t*, admin_err_t*) at 0xc5a068 (rethinkdb)\n6 [0x9b9dfe]: artificial_table_t::read_row(ql::env_t*, ql::datu
m_t, read_mode_t) at 0x9b9dfe (rethinkdb)\n7 [0x95dc55]: ql::get_selection_t::get() at 0x95dc55 (rethinkdb)\n8 [0x95a809]: ql::val_t::as_datum() const at 0x95a809 (rethinkdb)\n9 [0x75cbd2]: ql::query_cache_t::ref_t::run(ql::e
nv_t*, ql::response_t*) at 0x75cbd2 (rethinkdb)\n10 [0x75d69e]: ql::query_cache_t::ref_t::fill_response(ql::response_t*) at 0x75d69e (rethinkdb)\n11 [0x830c19]: rdb_query_server_t::run_query(ql::query_params_t*, ql::response_
t*, signal_t*) at 0x830c19 (rethinkdb)\n12 [0xaff0da]: query_server_t::handle(http_req_t const&, http_res_t*, signal_t*) at 0xaff0da (rethinkdb)\n13 [0xa61fbf]: routing_http_app_t::handle(http_req_t const&, http_res_t*, signa
l_t*) at 0xa61fbf (rethinkdb)\n14 [0xa61fbf]: routing_http_app_t::handle(http_req_t const&, http_res_t*, signal_t*) at 0xa61fbf (rethinkdb)\n15 [0xa6c96c]: http_server_t::handle_conn(scoped_ptr_t<linux_tcp_conn_descriptor_t>
const&, auto_drainer_t::lock_t) at 0xa6c96c (rethinkdb)\n16 [0xa6db02]: std::_Function_handler<void (scoped_ptr_t<linux_tcp_conn_descriptor_t>&), boost::_bi::bind_t<void, boost::_mfi::mf2<void, http_server_t, scoped_ptr_t<lin
ux_tcp_conn_descriptor_t> const&, auto_drainer_t::lock_t>, boost::_bi::list3<boost::_bi::value<http_server_t*>, boost::arg<1>, boost::_bi::value<auto_drainer_t::lock_t> > > >::_M_invoke(std::_Any_data const&, scoped_ptr_t<lin
ux_tcp_conn_descriptor_t>&) at 0xa6db02 (rethinkdb)\n17 [0x9f91cc]: linux_nonthrowing_tcp_listener_t::handle(int) at 0x9f91cc (rethinkdb)\n18 [0xa11b7e]: coro_t::run() at 0xa11b7e (rethinkdb)
```

This server never crashes, so I haven't really looked into the log_file very often ;-)

**Comment by danielmewes** [comment]
Thanks for the bug report @karlgrz . Sorry you had to run into this.

I just tried and can reproduce the crash with that query on a completely empty single server. We'll look into it.

**Comment by karlgrz** [comment]
No worries, @danielmewes, thanks for the lightning fast response, as always!

**Comment by mglukhovsky** [comment]
@karlgrz: I'd recommend that you run at least three servers in your cluster, so you can take advantage of [automatic failover](https://rethinkdb.com/docs/failover/) (at least three servers are required) to minimize downtime in the case a server goes down.

**Comment by karlgrz** [comment]
@mglukhovsky agreed wholeheartedly, we're only running two at the moment due to cost. We plan on adding a third node soon.

**Comment by VeXocide** [comment]
In CR 3450 by @Tryneus

**Comment by VeXocide** [comment]
Merged into `next` and `v2.2.x` via commit e3a4c81927ca7f5b025cdb71baef24f565fece4f and 3e440a4c4d18b41591d18990afb9f1a06245d28f respectively.

**Comment by karlgrz** [comment]
@VeXocide awesome, thanks for the incredibly fast turn around!
